assets/js/modules/analytics-4/utils/withCustomDimensions.js

```diff
@@ -0,0 +1,374 @@
+				return (
+					<MetricTileWrapper
+						infoTooltip={ tileInfoTooltip }
+						loading
+						moduleSlug="analytics-4"
+						title={ tileTitle }
+						Widget={ Widget }
+					/>
+				);
+			}
+
+			// Show error states.
+			if ( !! customDimensions && ! loading ) {
+				if ( !! customDimensionsCreationErrors.length ) {
+					// Handle permissions error encountered while creating
+					// custom dimensions.
+					if (
+						customDimensionsCreationErrors.some(
+							isInsufficientPermissionsError
+						)
+					) {
+						return (
+							<MetricTileError
+								title={ __(
+									'Insufficient permissions',
+									'google-site-kit'
+								) }
+								{ ...commonErrorProps }
+							>
+								<div className="googlesitekit-report-error-actions">
+									<span className="googlesitekit-error-retry-text">
+										{ createInterpolateElement(
+											__(
+												'Permissions updated? <a>Retry</a>',
+												'google-site-kit'
+											),
+											{
+												a: (
+													<Link
+														onClick={
+															handleCreateCustomDimensions
+														}
+													/>
+												),
+											}
+										) }
+									</span>
+									<span className="googlesitekit-error-retry-text">
+										{ createInterpolateElement(
+											__(
+												'You’ll need to contact your administrator. <a>Learn more</a>',
+												'google-site-kit'
+											),
+											{
+												a: (
+													<Link
+														href={ helpLink }
+														external
+													/>
+												),
+											}
+										) }
+									</span>
+								</div>
+							</MetricTileError>
+						);
+					}
+
+					// Handle generic errors encountered while creating
+					// custom dimensions.
+					return (
+						<MetricTileError
+							title={ __(
+								'Analytics update failed',
+								'google-site-kit'
+							) }
+							{ ...commonErrorProps }
+						>
+							<div className="googlesitekit-report-error-actions">
+								<Button
+									onClick={ handleCreateCustomDimensions }
+								>
+									{ __( 'Retry', 'google-site-kit' ) }
+								</Button>
+								<span className="googlesitekit-error-retry-text">
+									{ createInterpolateElement(
+										__(
+											'Retry didn’t work? <a>Learn more</a>',
+											'google-site-kit'
+										),
+										{
+											a: (
+												<Link
+													href={ helpLink }
+													external
+												/>
+											),
+										}
+									) }
+								</span>
+							</div>
+						</MetricTileError>
+					);
+				}
+
+				if ( ! hasCustomDimensions ) {
+					return (
+						<MetricTileError
+							title={ __( 'No data to show', 'google-site-kit' ) }
+							{ ...commonErrorProps }
+						>
+							<div className="googlesitekit-report-error-actions">
+								<Button
+									onClick={ handleCreateCustomDimensions }
+								>
+									{ __( 'Update', 'google-site-kit' ) }
+								</Button>
```

Comment:
The same here. How does `handleCreateCustomDimensions` help with updating custom dimensions? It will redirect users to the permissions granting scope. In my understanding we should just reset resolution for the `getReport` selector, right?